src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -4040,7 +4001,7 @@ public override BoundNode VisitAsOperator(BoundAsOperator node)
                 }
                 else
                 {
-                    SetUnknownResultNullability();
+                    this.State.ResultIsNotNull = null;
```

Comment:
`SetUnknownResultNullability` has a couple of additional checks that are already performed above. Setting `ResultIsNotNull` directly matches the other cases where this property is set within the same checks.